### PR TITLE
Work in progress:

### PR DESCRIPTION
**Pull request by IsmailM**
- Add install script
- Update rake method for packaging
- Add subcommands
- Include web app inside GV and standalone packages
- add csv output
- add --select-single-best
- allow users to choose a subset of the output (i.e. just HTML etc.)
- allow users to specify output folder

### CODE REVIEW

**Comment by yeban** [comment]
Failed to run genevalidator:

```
~/wurmlab/genevalidator dev 25s
❯ bundle
Fetching gem metadata from http://rubygems.org/..............
Resolving dependencies...
Bundler could not find compatible versions for gem "genevalidatorapp":
  In Gemfile:
    genevalidator was resolved to 1.7.2, which depends on
      genevalidatorapp (= 2.0)

Could not find gem 'genevalidatorapp (= 2.0)', which is required by gem 'genevalidator', in any of the sources.

Bundler could not find compatible versions for gem "prawn":
  In Gemfile:
    genevalidator was resolved to 1.7.2, which depends on
      statsample (= 2.1.0) was resolved to 2.1.0, which depends on
        reportbuilder (~> 1.4) was resolved to 1.4.2, which depends on
          prawn-svg (~> 0.9.1) was resolved to 0.9.1.11, which depends on
            prawn (>= 0.8.4)

    genevalidator was resolved to 1.7.2, which depends on
      statsample (= 2.1.0) was resolved to 2.1.0, which depends on
        reportbuilder (~> 1.4) was resolved to 1.4.2, which depends on
          prawn (~> 0.8.4)

# Install from RubyGems to get all the dependencies
~/wurmlab/genevalidator dev
❯ gem install genevalidator
[...]
Successfully installed genevalidator-1.6.12
1 gem installed 

# Now run
~/wurmlab/genevalidator dev
❯ bin/genevalidator
bin/genevalidator:382: can't find string "MSG" anywhere before EOF
bin/genevalidator:358: syntax error, unexpected end-of-input
```

**Comment by IsmailM** [comment]
yep need to push an updated Genevalidatorapp to rubygems - but need to test my changes first

- you might be able to install the updated GVapp locally, and then run a bundle install --local... 
- but still need to test GVapp - so cannot guarantee it will work as is atm

Also need to create and release new binaries....

**Comment by IsmailM** [comment]
I think everything should be working:

To Install:

```bash
sh -c "$(curl -fsSL https://raw.githubusercontent.com/wurmlab/genevalidator/dev/install.sh)"

cd genevalidator/exemplar_data

# uses local Swissprot by default
genevalidator -n 4 -i protein_data.fa

# Or to use swissprot -remote 
genevalidator -d "swissprot -remote" -n 4 -i protein_data.fa

```

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+GV_URL=$(curl -s https://api.github.com/repos/wurmlab/genevalidator/releases/latest \
+  | grep browser_download_url \
+  | grep -i $PLATFORM \
+  | cut -d '"' -f 4)
+
+echo >&2 "==> Installing GeneValidator to:"
+echo >&2 "    ${INSTALL_DIR}"
+
+mkdir "${INSTALL_DIR}"
+curl -sSL "$GV_URL" | tar zxf - -C "${INSTALL_DIR}" --strip-components 1
```
Comment (yeban) [inline install.sh]: As a user, I would like to see download progress (i.e., curl's output)

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  echo >&2 '==> No profile files were found.'
+  echo >&2 '    Please create one and add the following line to that file:'
+  echo >&2
+  echo >&2 '    export PATH="'"${INSTALL_DIR}"'/bin:${PATH}"'
+else
+  echo >&2 'export PATH="'"${INSTALL_DIR}"'/bin:${PATH}"' >> "${DOT_FILE}"
+  echo >&2
+  echo >&2 "==> Added GeneValidator to your PATH in ~/.zshrc"
+  echo >&2
+  echo >&2 "==> To start using GeneValidator you need to run \`source ~/.zshrc\`"
```
Comment (yeban) [inline install.sh]: "Run \`genevalidator -h\` in a new window to get started."

Comment (IsmailM) [reply install.sh]: Thanks - have added that between line 84 and 85. Should we remove lines 85-87 (i.e. the part about running `source $dotfile` etc.)?

Comment (yeban) [reply install.sh]: Yeah, I think someone who minds opening a new terminal window will know how to reload env.

**Comment by yeban** [comment]
CSV output contains a JSON like string:

"{:idx=>1, :overall_score=>12, :definition=>""SiOBP1-mRNA [organism=Solenopsis invicta]"", :no_hits=>3 ..."

**Pull request merged**